datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/mirrormaker/TestKafkaMirrorMakerConnectorTask.java

```diff
@@ -1142,7 +1142,44 @@ public void testFlowControlDisabled() throws Exception {
         "did not shut down on time");
   }
 
-  @Test (enabled = false) // Test disabled since it has been flaky
+  @Test
+  public void testAutoPauseOnSendError() throws Exception {
+    String[] cookieTopics = new String[] {
+        "MacadamiaWhiteChocolateCookie",
+        "AlmondCookie",
+        "ChocolateChipCookie",
+        "SaltyCaramelCookie"
+    };
+
+    for (String topic: cookieTopics) {
+      createTopic(_zkUtils, topic);
+    }
+
+    // create a datastream to consume from topics ending in "Pizza"
+    Datastream datastream = KafkaMirrorMakerConnectorTestUtils.createDatastream("cookieStream", _broker, "\\w+Cookie");
+
+    DatastreamTaskImpl task = new DatastreamTaskImpl(Collections.singletonList(datastream));
+    MockDatastreamEventProducer datastreamProducer = new MockDatastreamEventProducer((r) -> true);
+    task.setEventProducer(datastreamProducer);
+
+    KafkaMirrorMakerConnectorTask connectorTask =
+        KafkaMirrorMakerConnectorTestUtils.createFlushlessKafkaMirrorMakerConnectorTask(task, true, 50, 100,
+            Duration.ofDays(1));
+    KafkaMirrorMakerConnectorTestUtils.runKafkaMirrorMakerConnectorTask(connectorTask);
+
+    for (String topic: cookieTopics) {
+      KafkaMirrorMakerConnectorTestUtils.produceEvents(topic, 1, _kafkaCluster);
+    }
+
+    Assert.assertTrue(PollUtils.poll(() -> connectorTask.getAutoPausedSourcePartitions().size() == cookieTopics.length, POLL_PERIOD_MS, POLL_TIMEOUT_MS),
+        "All topics should have had auto-paused partitions");
```

Comment:
can you print the list of topics in the auto pause list when printing the reason for failure? Will be easier to compare which topics were auto-paused and which weren't

Reply:
The code that does the auto pause already prints that. I don't think it will be useful.